### PR TITLE
[Enhancement] Disable DiskAndTabletLoadReBalancer in shared data mode

### PR DESCRIPTION
**Pull request by sduzh**
Does not run DiskAndTabletLoadReBalancer in shared data mode to reduce database lock contention.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by kevincai — APPROVED** [review]

**Review by gengjun-git — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java [hunk 1]
```diff
@@ -78,6 +79,9 @@ public class DiskAndTabletLoadReBalancer extends Rebalancer {
     @Override
     protected List<TabletSchedCtx> selectAlternativeTabletsForCluster(
             ClusterLoadStatistic clusterStat, TStorageMedium medium) {
+        if (RunMode.getCurrentRunMode() == RunMode.SHARED_DATA) {
```
[the remark this answers is not in the document]
Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/clone/DiskAndTabletLoadReBalancer.java]: Updated

**Review by kevincai — APPROVED** [review]

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by sduzh** [comment]
@Mergifyio rebase

**Comment by sduzh** [comment]
@Mergifyio rebase

**Review by Astralidea — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Pull request merged**